cmd/podman/containers/restore.go

```diff
@@ -137,6 +165,9 @@ func restore(cmd *cobra.Command, args []string) error {
 	if argLen < 1 && !restoreOptions.All && !restoreOptions.Latest && restoreOptions.Import == "" {
 		return errors.Errorf("you must provide at least one name or id")
 	}
+	if argLen > 1 && restoreOptions.Name != "" {
+		return errors.Errorf("--name can only be used with one checkpoint image")
+	}
```

Comment:
I think that all these checks should happen _before_ image inspect.  It works as is just fine but the code is easier to maintain when we validate input before using it.

Reply:
This is slightly complicated because before image inspect we don't know if the input is an image or a container.
- If it is a container, `--name` is not supported.
- If it is an image, it should be only one.